File: src/lessons/recursion/cons/NfirstEntity.py
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(nfirst(toRecListIfArray(param[0]), toRecListIfArray(param[1]))))

# BEGIN TEMPLATE
def nfirst(list, n):
# BEGIN SOLUTION
  if n == 0:
    return None
  return cons(list.head, nfirst(list.tail, n-1))
# END SOLUTION
# END TEMPLATE

File: src/lessons/recursion/cons/NfirstEntity.scala
package lessons.recursion.cons

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest
import lessons.recursion.cons.universe.ConsEntity

class NfirstEntity extends ConsEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(nfirst( param(0).asInstanceOf[Array[Int]].toList, param(1).asInstanceOf[Int] ).toArray ))
	  }
	}

	/* BEGIN TEMPLATE */
	def nfirst(l:List[Int], n:Int): List[Int] = {
	/* BEGIN SOLUTION */
  if (n == 0) return Nil
  else        return l.head :: nfirst(l.tail, n-1)
	/* END SOLUTION */
	}
	/* END TEMPLATE */
}
